mypy/subtypes.py

```diff
@@ -106,11 +110,43 @@ class SubtypeVisitor(TypeVisitor[bool]):
     def __init__(self, right: Type,
                  type_parameter_checker: TypeParameterChecker,
                  *, ignore_pos_arg_names: bool = False,
-                 ignore_declared_variance: bool = False) -> None:
+                 ignore_declared_variance: bool = False,
+                 ignore_promotions: bool = False) -> None:
         self.right = right
         self.check_type_parameter = type_parameter_checker
         self.ignore_pos_arg_names = ignore_pos_arg_names
         self.ignore_declared_variance = ignore_declared_variance
+        self.ignore_promotions = ignore_promotions
+        self._subtype_kind = SubtypeVisitor.build_subtype_kind(
+            type_parameter_checker=type_parameter_checker,
+            ignore_pos_arg_names=ignore_pos_arg_names,
+            ignore_declared_variance=ignore_declared_variance,
+            ignore_promotions=ignore_promotions)
+
+    @staticmethod
+    def build_subtype_kind(*,
+                           type_parameter_checker: TypeParameterChecker = check_type_parameter,
+                           ignore_pos_arg_names: bool = False,
+                           ignore_declared_variance: bool = False,
+                           ignore_promotions: bool = False) -> SubtypeKind:
+        return hash(('subtype',
+                     type_parameter_checker,
+                     ignore_pos_arg_names,
+                     ignore_declared_variance,
+                     ignore_promotions))
```

Comment:
Is the `hash` call really needed here? As I understand it will be used as a key (tuples can also be keys) so the hash will be called under the hood anyway. Or is it a performance trick?

Reply:
It was an attempt at optimization, but I forgot to test it before submitting this PR. After testing, it ended up making little-to-no difference, so I ended up just removing the hash.